db/version_set.cc

```diff
@@ -1542,10 +1595,15 @@ void LevelIterator::SetFileIterator(InternalIterator* iter) {
   }
 
   InternalIterator* old_iter = file_iter_.Set(iter);
-  // Since this is a new table iterator, no need to call Prepare() if
-  // scan_opts_ is null
   if (iter && scan_opts_) {
-    file_iter_.Prepare(scan_opts_);
+    if (file_to_scan_opts_.get() &&
+        file_to_scan_opts_->find(file_index_) != file_to_scan_opts_->end()) {
+      const std::vector<ScanOptions>& opts =
+          file_to_scan_opts_->at(file_index_);
+      file_iter_.Prepare(&opts);
```

Comment:
Are we calling Prepare here because BlockBasedTableIterator::Prepare is still synchronous at this point? Once async IO is implemented there, then I guess we will allocate all the table iterators upfront and call Prepare on them in LevelIterator::Prepare?